### PR TITLE
chore(bevfusion): update parameters for improved bevfusion-cl training

### PR DESCRIPTION
**Pull request by SamratThapa120**
This pull request introduces significant improvements and updates to the BEVFusion-CL base and offline configurations, focusing on data preprocessing, augmentation, model configuration, and documentation. The changes enhance training stability, improve augmentation consistency, and update documentation to reflect the latest evaluation results.

**Key changes include:**

### Data Augmentation and Preprocessing
- Refactored the `sample_augmentation` method in `transforms_3d.py` to handle scalar `resize_lim` values, ensuring consistent resizing behavior and more robust augmentation during training and testing. Rotation now uses bicubic resampling to reduce artifacts. (`projects/BEVFusion/bevfusion/transforms_3d.py`) [[1]](diffhunk://#diff-a40de9d976eee70a128aac7ca0e89a6a67994ff4759ca57700b6526f9a571c29R27-R32) [[2]](diffhunk://#diff-a40de9d976eee70a128aac7ca0e89a6a67994ff4759ca57700b6526f9a571c29L38-R43) [[3]](diffhunk://#diff-a40de9d976eee70a128aac7ca0e89a6a67994ff4759ca57700b6526f9a571c29L55-R60)
Screenshot 2026-01-19 164301

- Updated the `ImageAug3D` pipeline in both training and testing to use a maximum resize limit scalar `resize_lim=0.02`, allowing training with images with various aspect ratios. (`projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py`, `projects/BEVFusion/configs/t4dataset/BEVFusion-CL/bevfusion_camera_lidar_voxel_second_secfpn_4xb8_base.py`) [[1]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2L178-R165) [[2]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2L288-R241) [[3]](diffhunk://#diff-2065d20a0e2a9df990c666bac1f49acdddc8967ca37259c903ead53bc3b5ed1bL160-R164) [[4]](diffhunk://#diff-2065d20a0e2a9df990c666bac1f49acdddc8967ca37259c903ead53bc3b5ed1bL243-R240)

### Model and Config Updates
- Major overhaul of the BEVFusion-CL-offline config: increased `train_gpu_size`, adjusted `image_size`, `feature_size`, `dbound`, and other model parameters for better performance and scalability; added `filter_cfg` to filter frames with missing images; and enabled automatic learning rate scaling. (`projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py`) [[1]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2L11-R14) [[2]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2R23-R61) [[3]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2L95-R102) [[4]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2R116) [[5]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2R126-L169) [[6]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2R269-R293) [[7]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2R311-R313) [[8]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2R353) [[9]](diffhunk://#diff-5f20835e30585d2980b7d71c9a261aeae0a5372b53bc162925ce5916dc2025a2L459-L469)
- Updated the BEVFusion-CL base config with new `image_size`, `feature_size`, and `dbound` values, and enabled automatic learning rate scaling. (`projects/BEVFusion/configs/t4dataset/BEVFusion-CL/bevfusion_camera_lidar_voxel_second_secfpn_4xb8_base.py`) [[1]](diffhunk://#diff-2065d20a0e2a9df990c666bac1f49acdddc8967ca37259c903ead53bc3b5ed1bL38-R38) [[2]](diffhunk://#diff-2065d20a0e2a9df990c666bac1f49acdddc8967ca37259c903ead53bc3b5ed1bL97-R101) [[3]](diffhunk://#diff-2065d20a0e2a9df990c666bac1f49acdddc8967ca37259c903ead53bc3b5ed1bL425-R422)

### Documentation and Evaluation Results
- Added a new documentation page summarizing the deployed BEVFusion-CL-offline base/2.X model, including training and evaluation details, metrics, and links to resources. (`projects/BEVFusion/docs/BEVFusion-CL-offline/v2/base.md`)
- Updated the BEVFusion-CL base documentation to include results for a new evaluation split (C). (`projects/BEVFusion/docs/BEVFusion-CL/v2/base.md`)

These changes collectively improve the robustness, reproducibility, and clarity of the BEVFusion-CL and BEVFusion-CL-offline pipelines, and provide up-to-date documentation for users and collaborators.

Improvement in bevfusion-CL base/2.0.0 before and after the changes

| Eval range: 120m                | mAP  | car  | truck | bus  | bicycle | pedestrian |
| --------------------------------| ---- | ---- | ----- | ---- | ------- | ---------- |
| BEVFusion-CL base/2.0.0 (B)     | **75.03** | 79.62 | 61.20 | **86.67** | 69.99 | **77.62**                |  
| BEVFusion-CL base/2.0.0 (C)     | **76.3**  | **80.50** | **61.90** |   85.90   | **74.70** | **78.70**    |
| BEVFusion-CL-offline base/2.0.0  (C) | **77.8**  | **87.30** | 61.60 | 85.90 | 73.20 | **80.90**     |

- BEVFusion-CL base/2.0.0 (B): Without intensity and training pedestrians without pooling
- BEVFusion-CL base/2.0.0 (C): Same as `BEVFusion-CL base/2.0.0 (B)` with improved image ROI cropping, and augmentation parameter fixes.

### CODE REVIEW

File: projects/BEVFusion/bevfusion/transforms_3d.py [hunk 1]
```diff
@@ -52,7 +57,7 @@ def img_transform(self, img, rotation, translation, resize, resize_dims, crop, f
         img = img.crop(crop)
         if flip:
             img = img.transpose(method=Image.FLIP_LEFT_RIGHT)
-        img = img.rotate(rotate)
+        img = img.rotate(rotate, resample=Image.BICUBIC)  # Default rotation introduces artifacts.
```
Comment (KSeangTan) [inline projects/BEVFusion/bevfusion/transforms_3d.py:60]: Would be nice if you have examples showing artifacts with the default rotation

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py [hunk 2]
```diff
@@ -92,16 +95,11 @@
...
-        # ybound=[-54.0, 54.0, 0.3],
-        # xbound=[-122.4, 122.4, 0.68],
-        # ybound=[-122.4, 122.4, 0.68],
+        feature_size=[72, 108],
         xbound=[-122.4, 122.4, 0.3],
         ybound=[-122.4, 122.4, 0.3],
         zbound=[-10.0, 10.0, 20.0],
-        # dbound=[1.0, 60.0, 0.5],
-        dbound=[1.0, 166.2, 1.4],
+        dbound=[1.0, 134, 1.4],
```
Comment (KSeangTan) [inline projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py:102]: Any reason we change it to `134`, I am thinking we should make the depth and bin size even smaller, and make sure it's evenly divided by the bin size?

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py [hunk 3]
```diff
@@ -456,14 +415,8 @@
     clip_grad=dict(max_norm=35, norm_type=2),
 )
 
-# Default setting for scaling LR automatically
-#   - `enable` means enable scaling LR automatically
-#       or not by default.
-#   - `base_batch_size` = (8 GPUs) x (4 samples per GPU).
-# auto_scale_lr = dict(enable=False, base_batch_size=32)
-auto_scale_lr = dict(enable=False, base_batch_size=train_gpu_size * train_batch_size)
+auto_scale_lr = dict(enable=True, base_batch_size=4)
```
Comment (KSeangTan) [inline projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py:418]: Keep the comment, and any reason we set it to `True`? Does it show any significant improvement/stability for training?

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py [hunk 4]
```diff
@@ -456,14 +415,8 @@
...
 
+# Only set if the number of train_gpu_size more than 1
 if train_gpu_size > 1:
     sync_bn = "torch"
-
-randomness = dict(seed=0, diff_rank_seed=False, deterministic=True)
```
Comment (KSeangTan) [inline projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py:469]: Any reason we delete it? I believe we need to keep it for reproducibility

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL/bevfusion_camera_lidar_voxel_second_secfpn_4xb8_base.py [hunk 5]
```diff
@@ -421,8 +418,7 @@
 #   - `enable` means enable scaling LR automatically
 #       or not by default.
 #   - `base_batch_size` = (8 GPUs) x (4 samples per GPU).
-# auto_scale_lr = dict(enable=False, base_batch_size=32)
-auto_scale_lr = dict(enable=False, base_batch_size=train_gpu_size * train_batch_size)
+auto_scale_lr = dict(enable=True, base_batch_size=32)
```
Comment (KSeangTan) [inline projects/BEVFusion/configs/t4dataset/BEVFusion-CL/bevfusion_camera_lidar_voxel_second_secfpn_4xb8_base.py:421]: Same above

Comment (KSeangTan) [reply projects/BEVFusion/configs/t4dataset/BEVFusion-CL/bevfusion_camera_lidar_voxel_second_secfpn_4xb8_base.py:421]: `base_batch_size` should be batch size per gpu, which should be `train_batch_size` according to here
https://github.com/open-mmlab/mmengine/blob/main/mmengine/_strategy/base.py#L696

File: projects/BEVFusion/docs/BEVFusion-CL/v2/base.md [hunk 6]
```diff
@@ -40,12 +41,47 @@
 | BEVFusion-CL base/2.0.0 (A)      | 69.99 | 79.41 | 64.64 | 83.58 | 67.03 | 55.28                |
 | BEVFusion-CL base/2.0.0 (B)     | 74.48 | 77.28 | 62.67 | 87.92 | 66.58 | 77.98                  |
 
+- BEVFusion-CL base/2.0.0 (A): Without intensity and training pedestrians with pooling pedestrians
+- BEVFusion-CL base/2.0.0 (B): Same as `BEVFusion-CL base/2.0.0 (A)` without pooling pedestrians
+- BEVFusion-CL base/2.0.0 (C): Same as `BEVFusion-CL base/2.0.0 (B)` with improved image ROI cropping, and augmentation parameter fixes.
```
Comment (KSeangTan) [inline projects/BEVFusion/docs/BEVFusion-CL/v2/base.md:46]: I suppose you meant 
`BEVFusion-CL base/2.0.0 (A)` is without pooling pedestrians, and `BEVFusion-CL base/2.0.0 (B)` with pooling pedestrians? Otherwise, the performance in pedestrians doesn't make sense to me

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py (hunk 2 above)
Comment (KSeangTan) [reply projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py:102]: bin size: `1.4` could be a little big too large

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL/bevfusion_camera_lidar_voxel_second_secfpn_4xb8_base.py [hunk 7]
```diff
@@ -94,14 +94,11 @@
...
-        feature_size=[32, 88],
-        # xbound=[-54.0, 54.0, 0.3],
-        # ybound=[-54.0, 54.0, 0.3],
+        feature_size=[48, 72],
         xbound=[-122.4, 122.4, 0.68],
         ybound=[-122.4, 122.4, 0.68],
         zbound=[-10.0, 10.0, 20.0],
-        # dbound=[1.0, 60.0, 0.5],
-        dbound=[1.0, 166.2, 1.4],
+        dbound=[1.0, 134, 1.4],
```
Comment (KSeangTan) [inline projects/BEVFusion/configs/t4dataset/BEVFusion-CL/bevfusion_camera_lidar_voxel_second_secfpn_4xb8_base.py:101]: Same above

**Review by KSeangTan — CHANGES_REQUESTED** [review]
LGTM overall, just need to tidy up documentation a little bit

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py (hunk 2 above)
Comment (SamratThapa120) [reply projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py]: The range we are predicting upto is 122.4m, 166.2-122.4=43.8m is too much buffer for vehicle length. I use 10m buffer for vehicle length, and to make it divisible by 1.4, I set it to 134.

File: projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py (hunk 4 above)
Comment (SamratThapa120) [reply projects/BEVFusion/configs/t4dataset/BEVFusion-CL-offline/bevfusion_camera_lidar_offline_voxel_second_secfpn_4xb8_base.py]: reverted change

File: projects/BEVFusion/docs/BEVFusion-CL/v2/base.md (hunk 6 above)
Comment (SamratThapa120) [reply projects/BEVFusion/docs/BEVFusion-CL/v2/base.md]: I just copied that part based on previous descriptions of A and B 
https://github.com/tier4/AWML/blob/main/projects/BEVFusion/docs/BEVFusion-CL/v2/base.md#bevfusion-cl-base200:~:text=BEVFusion%2DCL%20base/2.0.0%20(A)%3A%20Without,base/2.0.0%20(A)%20without%20pooling%20pedestrians

**Review by KSeangTan — APPROVED** [review]
LGTM!
